conda_build/build.py

```diff
@@ -1915,10 +1924,26 @@ def bundle_conda(output, metadata: MetaData, env, stats, **kw):
     if files:
         # Files is specified by the output
         # we exclude the list of files that we want to keep, so post-process picks them up as "new"
-        keep_files = {
-            os.path.normpath(pth)
-            for pth in utils.expand_globs(files, metadata.config.host_prefix)
-        }
+        if isinstance(files, dict):
+            # When file matching with include/exclude lists, only
+            # new_prefix_files are considered. Files in the PREFIX from other
+            # recipes (dependencies) are ignored
+            include = files.get("include", [])
+            exclude = files.get("exclude", [])
```

Comment:
See line 1818:

```python
files = output.get("files", [])
```

Reply:
My understanding is that `files.get("include")` may return `None`, but `include` *must* be a list. So we use the falsiness of `None` to replace it with `[]`. Equivalent to:

```
include = files['include'] if ('include' in files and files['include'] != None) else []
```

Reply:
You might have

```
- output: something
  files:
    include:
      - path
    exclude:
      - not-this-path  # [linux]
```

and then you'll have `files["exclude"] = None`. Not sure at what point we are ensuring that `include` is a list.

Under that circumstance `files.get("exclude", [])` will be `None` instead of the expected `[]`. Is that what you mean? I think we are saying the same 😬

Reply:
Yes. We are saying the same thing. I'm just trying to figure out if we should be doing something stronger than converting Falsy values into the empty list. Perhaps convert any non-list into a list? Something like:

```
include = files.get("include") if isinstance(files.get("include"), list) else []
```

Reply:
That might silence errors when users populate it with a dict or something accidentally. I think we can just play it simple with falsy promotions.